pkg/provisioner/conditionwaiter.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioner
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+)
+
+type conditionWaiter struct {
+	objectType string
+	objectName string
+	watcher    *conditionWatcher
+}
+
+func (w *conditionWaiter) Wait(ctx context.Context) (runtime.Object, error) {
+	var obj runtime.Object
+	var err error
+	namespace := w.watcher.namespace
+	switch w.objectType {
+	case azvolume:
+		obj, err = w.watcher.informerFactory.Disk().V1alpha1().AzVolumes().Lister().AzVolumes(namespace).Get(w.objectName)
+	case azvolumeattachment:
+		obj, err = w.watcher.informerFactory.Disk().V1alpha1().AzVolumeAttachments().Lister().AzVolumeAttachments(namespace).Get(w.objectName)
+	case azdrivernode:
+		obj, err = w.watcher.informerFactory.Disk().V1alpha1().AzDriverNodes().Lister().AzDriverNodes(namespace).Get(w.objectName)
+	}
+
+	waitEntry := w.watcher.getWaitEntry(w.objectName)
```

Comment:
Rather than using another lookup here, why not simply pass a pointer to the `waitEntry` on creation of the waiter?